### ISSUE TITLE
Question to Feodor2: Instead from Pale Moon, would you be able to directly fork from MOZILLA FIREFOX ESR ??

### ISSUE THREAD

**Issue by PLudkovski** [Feodor2/Mypal#320]
Many webpages are not loading properly, because we are not even on Version v78.X !
Pale Moon upgrades seem to be very late and lame.
People cannot waste years before certain websites are working again.

Therefore my question:
Instead from Pale Moon, would you be able to directly fork from MOZILLA FIREFOX ESR ??

**Comment by ClassicNick** [comment]
(source unverified) I've heard it's impossible to build Firefox 57+ (including 78 ESR) and target Windows XP/Vista due to the Rust compiler not being able to target those Operating Systems.

**Comment by PLudkovski** [comment]
@ClassicNick
Hello, can you please ask Graydon Hoare, the founder and inventor of Rust,
whether there really is absolutely no way around, and else if he could give us some insight and advise on
how to compile V78.X for it to run on XP and 2003 ??
His e-mail contact is: graydon@pobox.com
His Github profile is: https://github.com/graydon

**Comment by PLudkovski** [comment]
@ClassicNick
You can also try to ask ChrisDenton from the Rust Compiler Team about how we could get a working V78.X for XP and 2003!
https://github.com/ChrisDenton

**Comment by jazei-AR** [comment]
basic Chinese for me... I'm trying to understand.... the ask is ...MP should be upgrade to modern FF
the response is .. impossible MP was made like old FF, this is the Border the limit....
¿is it wrong?
what www....html is bad showed in MP? for ex.... let me see them please for try to understand
thank you and good year!!!

**Comment by Feodor2** [comment]
Yes i plan build firefox for winxp, if i shall success it will be not mypal but another browser.

**Comment by Feodor2** [comment]
I plan to begin soon but do not know if it will be.

**Comment by Anbima** [comment]
Have you checked whether you can make another browser for Windows XP?
Chrome or Slimjet?
Slimjet is very fast...

**Comment by Feodor2** [comment]
No, i do not use chrome usually, and 360chrome already is.

**Comment by Anbima** [comment]
360chrome is only in bad english. Much is Chinese. And there is no German.

**Comment by roytam1** [comment]
> due to the Rust compiler not being able to target those Operating Systems

not only this, mozilla started removing XP support in gecko codebase since gecko-53.

**Comment by PLudkovski** [comment]
@roytam1
But i was thinking Mypal isn´t even using Gecko, but is using the Goanna rendering engine?

**Comment by roytam1** [comment]
> But i was thinking Mypal isn´t even using Gecko, but is using the Goanna rendering engine?

so referring to the topic, you said "fork from firefox ESR" it must not be esr52 since goanna4 forked from esr52 and added new features.
and for newer ESR, XP support has been ceased and related codepaths were removed.

**Comment by Feodor2** [comment]
Of course to adapt rust for winxp required and will be the first step.

Let Goanna title does not confuse you, there was Palemoon 27 era when developers change word 'gecko' to 'goanna' everywhere and in 28 they did not do such again sure was no point, will it be new different engine after changing words?

**Comment by AroKol78** [comment]
> Of course to adapt rust for winxp required and will be the first step.

very hard
type in the github search in your repository the word "winxp" , "win xp" and "XP_WIN"
and try to match new source code `https://github.com/mozilla/gecko-dev`

**Comment by PLudkovski** [comment]
Feodor2, please ask Mr. Graydon Hoare, he can help to create Rust-compiler for XP:
graydon@pobox.com OR graydon.hoare2@gmail.com
Please ask him !!
